update-go-version.sh

```diff
@@ -18,4 +18,7 @@ for f in .github/workflows/*.yml; do
     if [ $(yq  eval '.jobs.build.strategy.matrix | has("go")' "$f") == "true" ]; then
       yq eval --inplace ".jobs.build.strategy.matrix.go[0] = ${golang_base_version} | .jobs.build.strategy.matrix.go[0] style=\"single\"" "$f";
     fi
+    if [ $(yq  eval '.jobs.build-installer.strategy.matrix | has("go")' "$f") == "true" ]; then
```

Comment:
Ideally we'd be able to use something like `.jobs.*.strategy.matrix` instead of hardcoding the name of the jobs.

Reply:
If we are going to create a following for this I just realized the name for the images `images/*/Dockerfile` will not work for `build-e2e` and `build-integration` as I use the `Containerfile` as name for the file.

Reply:
An additional question how does dependabot fit in this upgrade process?